library/std/src/sys/pal/solid/os.rs

```diff
@@ -191,19 +191,19 @@ pub fn getenv(k: &OsStr) -> Option<OsString> {
     .flatten()
 }
 
-pub fn setenv(k: &OsStr, v: &OsStr) -> io::Result<()> {
+pub unsafe fn setenv(k: &OsStr, v: &OsStr) -> io::Result<()> {
     run_with_cstr(k.as_bytes(), &|k| {
         run_with_cstr(v.as_bytes(), &|v| {
             let _guard = ENV_LOCK.write();
-            cvt_env(unsafe { libc::setenv(k.as_ptr(), v.as_ptr(), 1) }).map(drop)
+            cvt_env(libc::setenv(k.as_ptr(), v.as_ptr(), 1)).map(drop)
```

Comment:
This should probably be converted to `unsafe_op_in_unsafe_function` style and keep the unsafe block here.

Reply:
Done.